### PR TITLE
new recipes for rocblas,hipblas,rocsolver

### PR DESCRIPTION
**Pull request by srekolam**
Rocblas is currently dependent on the following. llvm-devel and llvm-static will install into /usr/lib64/llvm7.0 .
yum install llvm-toolset-7-llvm-devel
yum install llvm-toolset-7-llvm-static
yum install zlib-devel

Append the llvm-amdgpu prefix path to the PATH . This will compile the tensile kernels.
export PATH=$PATH:/spack/opt/spack/linux-centos7-zen/gcc-7.3.1/llvm-amdgpu-3.5.0-xxxxxxxxxxxxx 
add to CMAKE_PREFIX_PATH=/usr/lib64/llvm7.0

### CODE REVIEW

File: packages/hipblas/package.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def cmake_args(self):
+        spec=self.spec
+        args = [
+                '-DHIP_COMPILER=clang',
+                '-DCMAKE_CXX_COMPILER={}/bin/hipcc'.format(self.spec['hip'].prefix),
+                '-DUSE_HIP_CLANG=ON',
+                '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=FALSE',
+                '-DCMAKE_PREFIX_PATH={};{}'.format(self.spec['comgr'].prefix,self.spec['llvm-amdgpu'].prefix),
+                '-DHIP_CLANG_INCLUDE_PATH={}/lib/clang/11.0.0/include'.format(self.spec['llvm-amdgpu'].prefix)
```
Comment (arjun-raj-kuppala) [inline packages/hipblas/package.py]: Can you change version of clang to get it dynamically...i have this code snippet for the same as part of rocrand which is lying in pull request.

File: packages/rocblas/package.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        args = [
+                '-DHIP_COMPILER=clang',
+                '-DCMAKE_CXX_COMPILER={}/bin/hipcc'.format(self.spec['hip'].prefix),
+                '-DUSE_HIP_CLANG=ON',
+		'-DTensile_COMPILER=hipcc',
+		'-DTensile_CODE_OBJECT_VERSION=V3',
+                '-DRUN_HEADER_TESTING=OFF',
+                '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=FALSE',
+                '-DCMAKE_PREFIX_PATH=/usr/lib64/llvm7.0;{}'.format(self.spec['llvm-amdgpu'].prefix),
+                '-DHIP_CLANG_INCLUDE_PATH={}/lib/clang/11.0.0/include'.format(self.spec['llvm-amdgpu'].prefix)
```
Comment (arjun-raj-kuppala) [inline packages/rocblas/package.py]: Can you change version of clang to get it dynamically...i have this code snippet for the same as part of rocrand which is lying in pull request.

File: packages/rocsolver/package.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    def cmake_args(self):
+        spec=self.spec
+        args = [
+                '-DHIP_COMPILER=clang',
+                '-DCMAKE_CXX_COMPILER={}/bin/hipcc'.format(self.spec['hip'].prefix),
+                '-DUSE_HIP_CLANG=ON',
+                '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=FALSE',
+                '-DCMAKE_PREFIX_PATH={};{}'.format(self.spec['comgr'].prefix,self.spec['llvm-amdgpu'].prefix),
+                '-DHIP_CLANG_INCLUDE_PATH={}/lib/clang/11.0.0/include'.format(self.spec['llvm-amdgpu'].prefix)
```
Comment (arjun-raj-kuppala) [inline packages/rocsolver/package.py]: Can you change version of clang to get it dynamically...i have this code snippet for the same as part of rocrand which is lying in pull request.

File: packages/hipblas/package.py (hunk 1 above)
Comment (srekolam) [reply packages/hipblas/package.py]: made the changes as per the review comments

File: packages/rocblas/package.py (hunk 2 above)
Comment (srekolam) [reply packages/rocblas/package.py]: made the changes as per the review comment

File: packages/rocsolver/package.py (hunk 3 above)
Comment (srekolam) [reply packages/rocsolver/package.py]: made the changes as per review comment

**Review by arjun-raj-kuppala — APPROVED** [review]

**Pull request merged**